Clarify alpha releases and features (#4306)
camunda/camunda-docs: docs/reference/release-policy.md

title: "Release policy"
---

Camunda 8 follows the [Camunda release policy](https://camunda.com/release-policy/) with the following specific clarifications.

:::info
You can find deprecation and support announcements on the [Announcements](announcements.md) page.
:::

![Stable and alpha channels when provisioning a cluster](./img/diagram-releases.png)

## Alpha features and releases

It is important to understand the different ways the term "alpha" is used in the context of Camunda releases and features.

### Alpha feature

> Refers to a feature or component released as an alpha version, in an early state for you to test and participate in development by sharing your feedback before the feature reaches [general availability (GA)](/docs/reference/alpha-features.md#general-availability-ga). You can choose whether to enable alpha features in your cluster. See [alpha features](/docs/reference/alpha-features.md).

Reviewer: Not all alpha features have to be toggled specifically.

```suggestion
Refers to a feature or component released as an alpha version, in an early state for you to test and participate in development by sharing your feedback before the feature reaches [general availability (GA)](/docs/reference/alpha-features.md#general-availability-ga). Some alpha features can / have to be toggled on for your cluster. See [alpha features](/docs/reference/alpha-features.md).
```

Author: Changed to: "Some alpha features require turning on for your cluster before you can use them."

An **Update available** notification is shown in Console, recommending that you update to the latest version.

![Console with notice to update the cluster in Camunda 8 SaaS](img/update-console.png)

As of Camunda 8.5.0, the generation naming scheme in Camunda 8 SaaS changed to no longer include the patch version.

- The naming scheme used for the Camunda 8.5 generations is `Camunda <Major>.<Minor>+gen<N>`, where `N` is incremented with every atomic change to the component version set.

- This decouples the generation name from the particular patch level of the components it contains, as some component versions like Connectors are decoupled from other components.

- You can learn about the particular component patch version changes in the update dialogue to the latest generation available.

#### Update or restart for critical issues

In our managed service, we reserve the right to force update or restart a cluster immediately and without notice in advance if there is a critical security or stability issue.

## Self-Managed

When a new Camunda 8 version is released, Camunda 8 Self-Managed enterprise customers are notified via email.

:::info

> Non-enterprise customers can stay up to date via [release blogs](https://camunda.com/blog/category/release-notes.md), [announcements](/reference/announcements.md), or releases on [GitHub](https://github.com/camunda) and [Docker Hub](https://hub.docker.com/u/camunda).

Reviewer: Link https://docs.camunda.io/docs/next/reference/release-notes/?

Author: Changed to https://camunda.com/blog/category/releases/

title: "Release policy"
---

Camunda 8 follows the [Camunda release policy](https://camunda.com/release-policy/) with the following specific clarifications.

:::info
You can find deprecation and support announcements on the [Announcements](announcements.md) page.
:::

![Stable and alpha channels when provisioning a cluster](./img/diagram-releases.png)

## Alpha features and releases

It is important to understand the different ways the term "alpha" is used in the context of Camunda releases and features.

### Alpha feature

Refers to a feature or component released as an alpha version, in an early state for you to test and participate in development by sharing your feedback before the feature reaches [general availability (GA)](/docs/reference/alpha-features.md#general-availability-ga). Some alpha features require turning on for your cluster before you can use them. See [alpha features](/docs/reference/alpha-features.md).

### Alpha release

> Refers to a release made available between minor versions that allows you to preview an upcoming minor version and the alpha features included (for example, 8.6.0-alpha1, 8.6.0-alpha2, and so on). Camunda strives to release this type of release on a monthly basis. To learn more about the alpha features included in each alpha release, see [release notes](/docs/reference/release-notes/release-notes.md).

Reviewer: ```suggestion
Refers to a release made available between minor versions that allows you to preview an upcoming minor version and the alpha features included (for example, `8.6.0-alpha1`, `8.6.0-alpha2`, and so on). Camunda strives to release this type of release on a monthly basis. To learn more about the alpha features included in each alpha release, see [release notes](/docs/reference/release-notes/release-notes.md).
```

:::

## SaaS provisioning

In Camunda 8 SaaS we differentiate between components that are part of a Camunda 8 cluster (cluster components), and components outside the cluster (non-cluster components).

### Cluster components

A cluster typically consists of the following components:

- [Zeebe](/docs/components/zeebe/zeebe-overview.md)
- [Operate](/docs/components/operate/operate-introduction.md)
- [Tasklist](/docs/components/tasklist/introduction-to-tasklist.md)
- [Optimize]($optimize$/components/what-is-optimize)

You can provision cluster components using one of two channels, following the [Camunda release policy](https://camunda.com/release-policy/).

![Stable and alpha channels when provisioning a cluster](./img/channels.png)

#### Stable channel

You can use the **Stable** channel to access [general availability](/docs/reference/alpha-features.md#general-availability-ga) features for cluster components.

- Provides the latest feature and patch releases ready for most users at minimal risk.
- Releases follow semantic versioning and can be updated to the next minor or patch release without data loss.

> - On the stable channel, the last three supported minor versions are made available for provisioning.

Reviewer: I think the reality is that we provide all supported minor versions (ie. see end of [maintenance](https://docs.camunda.io/docs/next/reference/release-notes/) => this can be 3, or 4 depending on how many releases we did) - but not 100% sure. @ultraschuppi can you clarify please?

```suggestion
- On the stable channel, the last three supported minor versions are made available for provisioning.
```

Reviewer 2: yep, supported minors it is

Author: ```suggestion
- On the stable channel, all supported minor versions are made available for provisioning.
```
@MaxTru should we change it to this?

Reviewer: Yes - I think this makes sense. Thanks
